oeps/oep-0007.rst

```diff
@@ -0,0 +1,462 @@
+`b'byte'` literals throughout. Bare native-string literals should be used
+sparingly, and explicitly called out as intentional usages.  This "calling out"
+can be enforced by installing the `caniusepython3` pylint extension, which will
+flag a warning (`native-string`) on such uses.  A native string would then be
+instantiated as:
+
+    native = 'string'  # pylint: disable=native-string
+
+This version creates noisier code than Option 2, below, but makes it easier to
+incrementally migrate large files, without introducing breaking changes.
+
+Builtins
+--------
+
+To support changing functionality in builtin Python commands, we recommend
+using the functionality provided by the chosen compatibility library for your
+project.
+
+In the `future` library, these look like:
+
+   from builtins import object, range, str, bytes  # pylint: disable=redefined-builtins
+
+The futurize script (phase 2) should add these imports where needed, but the
+pylint pragma will (for now) need to be added manually.
+
+The Python standard library has been shuffled around a bit in the move to
+Python 3.  Future provides a few methods to manage this.  For packages in
+Python 3 that use a name that was not used in Python 2, installing future
+allows you to just use the Python 3 name of the package.  If the name was
+already used in Python 2, the new version can be installed from `future.moves`
+or `future.backports`.
+
+Do not use the provided `futures.stdlib.install_aliases()`.  It monkey-patches
+the standard library, and makes it more difficult to iteratively migrate
+different parts of the codebase.
+
+With `six`, the recommended behavior is to use the default builtin for object,
+but to use `six.text_type`, `six.binary_type`.  Most other changed
+functionality is described in the list of renames under
+[`six.moves`](https://pythonhosted.org/six/#module-six.moves) in the
+documentaion.
+
+Dictionaries and iterables views
+--------------------------------
+
+Instead of using `d.iterkeys()`, use `future.utils.viewkeys(d)` or
+`six.viewkeys(d)`.  If you need a list, use `list(*.viewkeys(d))`.
+Other similar functions exist for `itervalues()` and `iteritems()`.  These
+changes cannot be made cleanly in the import headers, and will require more
+work to change after the fact.  This can be avoided in some cases by iterating
+directly over the dict object.  Instead of using:
+
+    for key, value in six.viewitems(d):
+        print(key, value)
+
+You could do:
+
+    for key in d:
+        value = d[key]
+        print(key, value)
+
+Packaging
+---------
+
+All packages should maintain the proper trove classifiers for the versions of
+Python they support.
+
+In the following recommendations, the major version classifiers comprise:
+
+    Programming Language :: Python :: 2
+    Programming Language :: Python :: 2 :: Only
+    Programming Language :: Python :: 3
+    Programming Language :: Python :: 3 :: Only
```

Comment:
The odd-numbered lines from these blocks are being omitted from the output; the colons are probably throwing off the formatting somehow